src/libs/ascent/runtimes/expressions/ascent_expression_filters.cpp

```diff
@@ -3554,6 +3556,77 @@ BinByValue::execute()
   set_output<conduit::Node>(output);
 }
 
+//-----------------------------------------------------------------------------
+AddFields::AddFields() : Filter()
+{
+  // empty
+}
+
+//-----------------------------------------------------------------------------
+AddFields::~AddFields()
+{
+  // empty
+}
+
+//-----------------------------------------------------------------------------
+void
+AddFields::declare_interface(Node &i)
+{
+  i["type_name"] = "add_fields";
+  i["port_names"].append() = "field1";
+  i["port_names"].append() = "field2";
+  i["port_names"].append() = "num_fields";
+  i["port_names"].append() = "output_name";
+  i["output_port"] = "true";
+}
+
+//-----------------------------------------------------------------------------
+bool
+AddFields::verify_params(const conduit::Node &params, conduit::Node &info)
+{
+  info.reset();
+  bool res = true;
+  return res;
+}
+
+//-----------------------------------------------------------------------------
+void
+AddFields::execute()
+{
+	std::cerr << " IN AddFields experssions_filters" << std::endl;
+
+  conduit::Node &n_fields = *input<Node>("num_fields");
+  conduit::Node &n_field1 = *input<Node>("field1");
+  conduit::Node &n_field2 = *input<Node>("field2");
+  conduit::Node &n_output = *input<Node>("output_name");
+  int32 num_fields = n_fields["value"].to_int32();;
+  const std::string field1 = n_field1["value"].as_string();
+  const std::string field2 = n_field2["value"].as_string();
+  const std::string out_field = n_output["value"].as_string();
+  std::cerr << "fields size; num_fields: " << num_fields<< std::endl;
+  std::cerr << "field1: " << field1 << " field2: " << field2 << " output_field: " << out_field << std::endl;
+
+  DataObject *data_object =
+    graph().workspace().registry().fetch<DataObject>("dataset");
+  conduit::Node *dataset = data_object->as_low_order_bp().get();
+
+  if(!is_scalar_field(*dataset, field1) || !is_scalar_field(*dataset, field2))
+  {
+    ASCENT_ERROR("AddFields: input fields for AddFields must be scalar fields. "
+                 "Invalid fields: '"
+                 << field1 << " or " << field2 << ".");
+  }
+
+  derived_field_add(*dataset, field1, field2, out_field);
+  std::cerr << "dataset after add" << std::endl;
+  dataset->print();
+
+//  resolve_symbol_result(graph(), output, this->name());
+  //set_output<conduit::Node>(output);
+}
```

Comment:
Now that I am adding the fields directly to the dataset, what should this output be? I believe this is the output that gets written to the sessions file. I could output the new field? Or just metadata, like fields added and output field?

Reply:
Good question - the case were we paint the binning onto the mesh is the closest to this case, do we have a precedent there that makes sense?

Reply:
Good suggestion, I think that makes sense. I'll base it off that example.